runtime/parser/errors.go

```diff
@@ -91,6 +91,60 @@ func (e *SyntaxError) Error() string {
 	return e.Message
 }
 
+// SyntaxErrorWithSuggestedFix
+
+type SyntaxErrorWithSuggestedFix struct {
```

Comment:
Maybe rename to `SyntaxErrorWithSuggestReplacement` to indicate that the fix is a replacement, and not a removal or insertion.